src/verifiable/converters.ts

```diff
@@ -0,0 +1,38 @@
+import { JWS } from 'jose'
+
+import { Address, VerifiedAddress } from './verifiable-payid'
+
+/**
+ * Converts a GeneralJWS to a Verified Address.
+ *
+ * @param jws - The JWS to convert.
+ * @returns The address.
+ * @throws Error if the JWS does not have the required properties.
+ */
+export function convertToVerifiedAddress(jws: JWS.GeneralJWS): VerifiedAddress {
+  const verified: VerifiedAddress = {
+    payload: jws.payload,
+    signatures: jws.signatures.map((value) => {
+      if (value.protected) {
+        return {
+          protected: value.protected,
+          signature: value.signature,
+        }
+      }
+      throw Error('missing protected property')
+    }),
+  }
+  return verified
+}
```

Comment:
What is the role / purpose of this function?

Reply:
We map the PayID information including JWS stuff with our own types but the jose library has it's own types for signatures. The VerifiedAddress type mirrors the structure of the jose JWS.Recipient type. I needed to convert the Recipient to a VerifiedAddress in order to construct a full PaymentInformation object. If we just said, screw it, we'll use the Jose type instead of VerifiedAddress this wouldn't be necessary. But then we'd be coupling our types to the Jose library which we probably don't want to do.

Reply:
Worth adding some more explanation to the JSDoc for this function?